tests/test_calculator.hh

```diff
@@ -715,72 +748,114 @@ namespace rascal {
+     *         atom (not a ghost or periodic image).
      */
-    PairRef_t swap_pair_ref(const PairRef_t & pair_ref) {
+    std::vector<PairRefKey_t> swap_pair_ref(const PairRef_t & pair_ref) {
+      auto center_manager{extract_underlying_manager<0>(structure_manager)};
+      auto atomic_structure{center_manager->get_atomic_structure()};
       // Get the atom index to the corresponding atom tag
-      size_t access_index = structure_manager->get_atom_index(pair_ref.back());
+      size_t access_index{structure_manager->get_atom_index(pair_ref.back())};
       auto new_center_it{structure_manager->get_iterator_at(access_index)};
       // Return cluster ref at which the iterator is currently pointing
       auto && new_center{*new_center_it};
-      // Iterate until (j,i) is found
+      size_t i_index{structure_manager->get_atom_index(pair_ref.front())};
+
+      // Find all (j, i') pairs
+      std::vector<PairRefKey_t> new_pairs;
       for (auto new_pair : new_center) {
-        if (new_pair.back() == pair_ref.front()) {
-          return new_pair;
+        size_t i_trial_index{
+            structure_manager->get_atom_index(new_pair.back())};
+        // Is this the i (old center) atom or any of its images?
+        if (i_trial_index == i_index) {
+          new_pairs.emplace_back(std::move(new_pair));
         }
       }
-      std::stringstream err_str{};
-      err_str << "Didn't find symmetric pair for pair (i=" << pair_ref.front()
-              << ", j=" << pair_ref.back() << ").";
-      throw std::range_error(err_str.str());
+      if (new_pairs.size() == 0) {
+        std::stringstream err_str{};
+        err_str << "Didn't find any pairs for pair (i=" << pair_ref.front()
+                << ", j=" << pair_ref.back()
+                << "); access index for j = " << access_index;
+        throw std::range_error(err_str.str());
+      }
+      return new_pairs;
     }
   };
 
   /**
    * Test fixture holding the gradient calculator and structure manager
    *
-   * Holds data (i.e. function values, gradient directions) and iterates through
-   * the list of centers
+   * Holds data (function values, gradient directions, verbosity) and iterates
+   * through the list of centers
    */
-  template <typename RepManager_t, class StructureManager_t>
-  class RepresentationManagerGradientFixture : public GradientTestFixture {
+  template <typename RepCalculator_t, class StructureManager_t>
+  class RepresentationCalculatorGradientFixture : public GradientTestFixture {
    public:
     using StdVector2Dim_t = std::vector<std::vector<double>>;
     using Calculator_t =
-        RepresentationManagerGradientCalculator<RepManager_t,
-                                                StructureManager_t>;
+        RepresentationCalculatorGradientProvider<RepCalculator_t,
+                                                 StructureManager_t>;
 
     static const size_t n_arguments = 3;
```

Comment:
not in this PR, but I would rather make static values `ALL_CAPS`

Reply:
Coding convention treats static and dynamic variables the same. If you think this is bad, raise an issue at the next group meeting.